common/djangoapps/util/milestones_helpers.py

```diff
@@ -351,12 +341,31 @@ def get_course_content_milestones(course_id, content_id, relationship, user_id=N
     """
     if not settings.FEATURES.get('MILESTONES_APP', False):
         return []
-    from milestones import api as milestones_api
+
+    if user_id is None:
+        return milestones_api.get_course_content_milestones(course_id, content_id, relationship)
+
+    request_cache_dict = request_cache.get_cache(REQUEST_CACHE_NAME)
+    if user_id not in request_cache_dict:
+        request_cache_dict[user_id] = milestones_api.get_course_content_milestones(
+            course_key=course_id,
+            user={"id": user_id}
+        )
+    milestones_for_content = []
+    if relationship == "requires":
+        for milestone in request_cache_dict[user_id]:
+            if milestone["content_id"] == content_id and milestone["requirements"]:
+                milestones_for_content.append(milestone)
+    if relationship == "fulfills":
+        for milestone in request_cache_dict[user_id]:
+            if milestone["namespace"].contains(content_id) and milestone["requirements"]:
+                milestones_for_content.append(milestone)
```

Comment:
I'm not following what we are trying to do in the code being added here. It looks like we construct a `milestones_for_content` list, but then never use it. Can you help clarify?